### DISCUSSION TITLE
Cannot add foreign Key

### DISCUSSION THREAD

**Discussion by ItzikFekete** [sqlitebrowser/sqlitebrowser#3119] [General]
Details for the issue
--------------------

#### What did you do? 

I used PRAGMA to enable foreign key but I couldn't edit the table to create a foreign key. Iy came up with "refrence" ub the buttom but was not accepting my input in the "name column" of Constraints. 

#### What did you see instead?

Useful extra information
-------------------------
> DB4S v3.12.2 [built for i386-little_endian-ilp32] on Windows 10 (10.0) (winnt/10.0.19044) [x86_64]
> using SQLite Version 3.35.5
> and Qt 5.12.8

**Comment by chrisjlocke** [comment]
The dialog is a bit 'quirky' but should work OK.

![image](https://user-images.githubusercontent.com/7872218/182094762-40afbc5b-e379-41b6-be6f-1a0b3ca42f8e.png)

Lets assume you have table A and table B.  Table B has a foreign key to table A.

![image](https://user-images.githubusercontent.com/7872218/182094913-0ff7df82-64c5-4802-bd79-bb9557bee47e.png)

If I double-click the 'field1' line, I get the above edit controls in the 'Foreign key' part in the 'Edit table definition' dialog.

![image](https://user-images.githubusercontent.com/7872218/182095119-89830bd6-d20c-4d40-bfde-e5fb8ef70cc1.png)

Select the table

![image](https://user-images.githubusercontent.com/7872218/182095172-15b1953d-2bde-4a06-b89a-1976234c462c.png)

and the field.

![image](https://user-images.githubusercontent.com/7872218/182095228-94dde598-d9c9-47d1-bd8d-c90ea561f00c.png)

And the clause

![image](https://user-images.githubusercontent.com/7872218/182095573-1e0fa07b-d476-487d-9b41-3aa113f2251f.png)

The table definition now has this reference.

![image](https://user-images.githubusercontent.com/7872218/182095666-6b6fe192-13b8-4e1c-98fb-c9e7ca3abbae.png)

If I now add a new record

![image](https://user-images.githubusercontent.com/7872218/182095737-05344532-1ff9-4e0a-a64e-eca9fd5b8775.png)

DB4S shows a drop-down of available options.  Empty, as the tableA has no entries.

**Comment by chrisjlocke** [comment]
![image](https://user-images.githubusercontent.com/7872218/182096081-fc4311bf-80b8-4f07-9e24-e094b2652c82.png)

Trying to force a value in that field that doesn't exist in tableA.field1 will generate an error.
